control-plane/catalog/to-consul/resource.go

```diff
@@ -999,3 +1025,18 @@ func (t *ServiceResource) isIngressService(key string) bool {
 func consulHealthCheckID(k8sNS string, serviceID string) string {
 	return fmt.Sprintf("%s/%s", k8sNS, serviceID)
 }
+
+// Calculates the passing service weight
+func getServiceWeight(weight string) (int, error) {
+	// error validation if the input param is a number
+	weightI, err := strconv.Atoi(weight)
+	if err != nil {
+		return -1, err
+	}
+
+	if weightI <= 1 {
+		return -1, fmt.Errorf("expecting the service annotation %s value to be greater than 1", annotationServiceWeight)
```

Comment:
Maybe we can add some additional checks here. Can the service weight be greater than 100? if it can, ignore this comment, but just wanted to ensure we have completeness here wrt failing for all invalid service weight!

Reply:
Hey thanks for sharing the thoughts. I couldn't find any documentation which mentions the weight to be percentage. To start with all the service default weight is set to 1 hence it is 1:1 which can be considered as equal ratio diversion. I guess in that case it is allowed to go beyond 100.

I have seen some example on stackoverflow, users configuring it to 10000:1, not sure if it is valid scenario unless someone wants to do super safe canary test.

Reply:
Cool! thanks for investigating this!